inst/include/interface/rcpp/rcpp_objects/rcpp_recruitment.hpp

```diff
@@ -15,219 +15,317 @@
+    }
 
-  virtual uint32_t get_id() { return this->id; }
+    virtual uint32_t get_id() {
+        return this->id;
+    }
 
-  virtual double evaluate(double spawners, double ssbzero) {
-    fims::SRBevertonHolt<double> BevHolt;
+    virtual double evaluate(double spawners, double ssbzero) {
+        fims::SRBevertonHolt<double> BevHolt;
 
-    BevHolt.steep = this->steep.value;
-    BevHolt.rzero = this->rzero.value;
-    return BevHolt.evaluate(spawners, ssbzero);
-  }
-
-  /** @brief this adds the parameter values and derivatives to the TMB model
-   * object */
-  virtual bool add_to_fims_tmb() {
-    // base model
-    std::shared_ptr<fims::Information<TMB_FIMS_REAL_TYPE> > d0 =
-        fims::Information<TMB_FIMS_REAL_TYPE>::GetInstance();
-
-    std::shared_ptr<fims::SRBevertonHolt<TMB_FIMS_REAL_TYPE> > b0 =
-        std::make_shared<fims::SRBevertonHolt<TMB_FIMS_REAL_TYPE> >();
-
-    // set relative info
-    b0->id = this->id;
-    b0->steep = this->steep.value;
-    if (this->steep.estimated) {
-      if (this->steep.is_random_effect) {
-        d0->RegisterRandomEffect(b0->steep);
-      } else {
-        d0->RegisterParameter(b0->steep);
-      }
-    }
-    b0->rzero = this->rzero.value;
-    if (this->rzero.estimated) {
-      if (this->rzero.is_random_effect) {
-        d0->RegisterRandomEffect(b0->rzero);
-      } else {
-        d0->RegisterParameter(b0->rzero);
-      }
-    }
-    b0->log_sigma_recruit = this->log_sigma_recruit.value;
-    if (this->log_sigma_recruit.estimated) {
-      if (this->log_sigma_recruit.is_random_effect) {
-        d0->RegisterRandomEffect(b0->log_sigma_recruit);
-      } else {
-        d0->RegisterParameter(b0->log_sigma_recruit);
-      }
-    }
-    // add to Information
-    d0->recruitment_models[b0->id] = b0;
-
-    // first-order derivative
-    std::shared_ptr<fims::Information<TMB_FIMS_FIRST_ORDER> > d1 =
-        fims::Information<TMB_FIMS_FIRST_ORDER>::GetInstance();
-
-    std::shared_ptr<fims::SRBevertonHolt<TMB_FIMS_FIRST_ORDER> > b1 =
-        std::make_shared<fims::SRBevertonHolt<TMB_FIMS_FIRST_ORDER> >();
-
-    // set relative info
-    b1->id = this->id;
-    b1->steep = this->steep.value;
-    if (this->steep.estimated) {
-      if (this->steep.is_random_effect) {
-        d1->RegisterRandomEffect(b1->steep);
-      } else {
-        d1->RegisterParameter(b1->steep);
-      }
-    }
-    b1->rzero = this->rzero.value;
-    if (this->rzero.estimated) {
-      if (this->rzero.is_random_effect) {
-        d1->RegisterRandomEffect(b1->rzero);
-      } else {
-        d1->RegisterParameter(b1->rzero);
-      }
-    }
-    b1->log_sigma_recruit = this->log_sigma_recruit.value;
-    if (this->log_sigma_recruit.estimated) {
-      if (this->log_sigma_recruit.is_random_effect) {
-        d1->RegisterRandomEffect(b1->log_sigma_recruit);
-      } else {
-        d1->RegisterParameter(b1->log_sigma_recruit);
-      }
+        BevHolt.logit_steep = this->logit_steep.value;
```

Comment:
suggestion (non-blocking): we might want to consider throwing a warning if steepness is very close to but not quite 1.

Reply:
I suggest making the warning the same for estimable or non-estimable cases. Think about 10 years from now ... if it is a bad idea that a parameter is estimated close to one, it is probably a bad idea that someone is fixing it close to one as well.

Reply:
@kellijohnson-NOAA, my initial comment stated 'if steepness is close to 1 & estimable' but it should have read 'non-estimable.' Sorry for the confusion! Given a logit transformation, a steepness value estimated close to 1 will result in non-convergence, which is why we only considered only throwing an error for the non-estimable case.

Reply:
@ChristineStawitz-NOAA, should the warning be based on steepness near 1 opposed to logit steepness? Also I think we want to throw the warning when users want to fix the value (not estimated), correct?

Reply:
I removed the if is estimated flag. I also thought it was more likely users would accidentally set logit_steep to 1 than infinity, but we can add warnings for the other mistake in MQ